frontend/src/routes/App.tsx

```diff
@@ -14,14 +17,43 @@ function App() {
     navigate("/deposit")
   }
 
+  useEffect(() => {
+    // https://stackoverflow.com/questions/47879864/how-can-i-check-if-a-browser-supports-webassembly
+    const checkWasm = async () => {
+      try {
+        if (typeof WebAssembly === "object"
+          && typeof WebAssembly.instantiate === "function") {
+          const module = new WebAssembly.Module(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00));
+          if (!(module instanceof WebAssembly.Module)) {
+            throw new Error("Couldn't instantiate WASM Module")
+          }
+        } else {
+          throw new Error("No WebAssembly global object found")
+        }
+      } catch (e) {
+        console.error(e)
+        setWasmSupported(false);
+      }
+    }
+    checkWasm();
+
+  }, [])
+
   return (
     <div className="flex flex-col h-screen">
       <header className='p-8'>
         <img src={logo} className="App-logo" alt="logo" />
         <h2>You're probably looking for <a href="/tests">the tests</a></h2>
+        {wasmSupported ? <p>WASM works!</p> :
```

Comment:
We probably don't need anything to appear if it works.

Reply:
yeah but it's fun for now